wallet/src/repository/wallet/mod.rs

```diff
@@ -1292,6 +1252,79 @@ where
         Ok(None)
     }
 
+    // During wallet synchronization, generate external and internal addresses
+    // if transaction outputs are pointing to transient addresses
+    fn _sync_address_generation(
+        &self,
+        state: &mut State,
+        outputs: &[types::VttOutput],
+    ) -> Result<()> {
+        loop {
+            let (new_external_index, new_internal_index) = outputs.iter().fold(
```

Comment:
Is it possible to move the fold outside of the loop, or would that break everything?

Reply:
I think yes. For sure, there is a better way to do that address generation, but we valued simplicity over performance (giving the fact the little time we have).